sim/rogue/item_sets_phase_8.go

```diff
@@ -7,18 +7,114 @@ import (
+			rogue.applyScarletEnclaveDamage6PBonus()
+		},
+	},
+})
+
+// 2P
+// While Just a Flesh Wound is not active, your Backstab, Sinister Strike, Saber Slash, and Mutilate deal 10% increased damage per your active Poison or Bleed effect
+// afflicting the target, up to a maximum increase of 30%
+func (rogue *Rogue) applyScarletEnclaveDamage2PBonus() {
+	if rogue.HasRune(proto.RogueRune_RuneJustAFleshWound) {
+		return
+	}
+
+	label := "S03 - Item - Scarlet Enclave - Rogue - Damage 2P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	spellsModifiedBySetBonus := ClassSpellMask_RogueBackstab | ClassSpellMask_RogueSinisterStrike | ClassSpellMask_RogueSaberSlash | ClassSpellMask_RogueMutilateHit
+
+	damageMod := rogue.AddDynamicMod(core.SpellModConfig{
+		Kind:       core.SpellMod_DamageDone_Pct,
+		ClassMask:  spellsModifiedBySetBonus,
+		FloatValue: 1.0,
+	})
+
+	core.MakeProcTriggerAura(&rogue.Unit, core.ProcTrigger{
+		ActionID:       core.ActionID{SpellID: 1226843},
+		Name:           label,
+		Callback:       core.CallbackOnApplyEffects,
+		ClassSpellMask: spellsModifiedBySetBonus,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			totalBleedsAndPoisons := rogue.PoisonsActive[rogue.CurrentTarget.UnitIndex] + rogue.BleedsActive[rogue.CurrentTarget.UnitIndex]
+
+			// Only apply the damage mod up to 3 times for the 30% bonus maximum
+			damageMod.UpdateFloatValue(1 + 0.10*float64(min(3, totalBleedsAndPoisons)))
+			damageMod.Activate()
+		},
+	})
+}
+
+// 4P
+// Your Poison and autoattack critical strikes have a 10% chance to grant you a combo point. (Proc chance: 10%)
+func (rogue *Rogue) applyScarletEnclaveDamage4PBonus() {
+	label := "S03 - Item - Scarlet Enclave - Rogue - Damage 4P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	// Combo! spell that adds the combo point for the set bonus
+	comboPointMetrics := rogue.NewComboPointMetrics(core.ActionID{SpellID: 1226869})
+
+	// Add a combo point for poison tick crit
+	core.MakeProcTriggerAura(&rogue.Unit, core.ProcTrigger{
+		Name:           label,
+		ClassSpellMask: ClassSpellMask_RogueDeadlyPoisonTick | ClassSpellMask_RogueOccultPoisonTick | ClassSpellMask_RogueInstantPoison,
+		Callback:       core.CallbackOnPeriodicDamageDealt | core.CallbackOnSpellHitDealt,
+		Outcome:        core.OutcomeCrit,
+		ProcChance:     0.10,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			rogue.AddComboPoints(sim, 1, rogue.CurrentTarget, comboPointMetrics)
+		},
+	})
+
+	// Add a combo point for autoattack crit
+	core.MakeProcTriggerAura(&rogue.Unit, core.ProcTrigger{
+		Name:       label,
+		Callback:   core.CallbackOnPeriodicDamageDealt | core.CallbackOnSpellHitDealt,
+		Outcome:    core.OutcomeCrit,
+		ProcChance: 0.10,
+		ProcMask:   core.ProcMaskMeleeWhiteHit,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			rogue.AddComboPoints(sim, 1, rogue.CurrentTarget, comboPointMetrics)
+		},
```

Comment:
Oh I just realized these both share the same `Name`. I'm surprised that that isn't blowing up 🤔. Maybe it would be worth combining them into one custom aura

```suggestion
  procClassMasks := ClassSpellMask_RogueDeadlyPoisonTick | ClassSpellMask_RogueOccultPoisonTick | ClassSpellMask_RogueInstantPoison
  
  core.MakePermanent(rogue.RegisterAura(core.Aura{
    Label: label
    OnPeriodicDamageDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
      if result.DidCrit() && spell.Matches(procClassMasks) && sim.Proc(0.10, "Combo! proc") {
        rogue.AddComboPoints(sim, 1, rogue.CurrentTarget, comboPointMetrics)
      }
    },
    OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
      if result.DidCrit() && (spell.Matches(procClassMasks) || spell.ProcMask.Matches(core.ProcMaskMeleeWhiteHit) && sim.Proc(0.10, "Combo! proc") {
        rogue.AddComboPoints(sim, 1, rogue.CurrentTarget, comboPointMetrics)
      }
    }
  }))
```